backend/database/database.go

```diff
@@ -391,6 +394,43 @@ func (d *Database) ReviewsByVendorID(vendorID uuid.UUID) ([]Review, error) {
 	return result, rows.Err()
 }
 
+//query to find out if the vendor has been starred, it returns true if it is false if not
+func (d *Database) StarredVendorFavorite(VendorID uuid.UUID, ReviewsID uuid.UUID) (bool, error) {
+	var starred bool
+
+	if err := d.db.QueryRow("SELECT * FROM Reviews WHERE ID >= ?",
+		ReviewsID, VendorID).Scan(&starred); err != nil {
+
+		return false,
```

Comment:
The error may not be related to a nonexistent review. Therefore, this error message may become a source of confusion if the actual error is different. You should be returning the error like this: `return false, err`.